torchdata/dataloader2/communication/eventloop.py

```diff
@@ -64,8 +72,8 @@ def SpawnProcessForDataPipeline(multiprocessing_ctx, datapipe, call_on_process_i
 
 def SpawnThreadForDataPipeline(datapipe):
     r"""
-    Given a DataPipe, creates a copy of the DataPipe, starts a new Thread with DataPipeToQueuesLoop as target,
-    and return the process, req_queue, res_queue, thread_local_datapipe.
+    Given a DataPipe, creates a copy of the DataPipe, starts a new Thread with ``DataPipeToQueuesLoop`` as target,
+    and returns ``(process, req_queue, res_queue, new_copied_datapipe)``.
```

Comment:
Actually, I don't think we will need to since the Process version doesn't need to either.

Reply:
I think since it is using Thread (the Process version is above it), we need copy once for each thread to avoid sharing of state? Please correct me if I'm wrong.

Reply:
I see. Makes sense to me